client/renderSDL/SDL_Extensions.cpp

```diff
@@ -804,7 +804,10 @@ void CSDL_Ext::fillRect( SDL_Surface *dst, const Rect & dstrect, const SDL_Color
 	SDL_Rect newRect = CSDL_Ext::toSDL(dstrect);
 
 	uint32_t sdlColor = SDL_MapRGBA(dst->format, color.r, color.g, color.b, color.a);
-	SDL_FillRect(dst, &newRect, sdlColor);
+	SDL_Surface * tmp = SDL_CreateRGBSurface(0, newRect.w, newRect.h, dst->format->BitsPerPixel, dst->format->Rmask, dst->format->Gmask, dst->format->Bmask, dst->format->Amask);
+	SDL_FillRect(tmp, NULL, sdlColor);
+	SDL_BlitSurface(tmp, NULL, dst, &newRect);
+	SDL_FreeSurface(tmp);

```

Comment:
Likely cause of regression:
![image](https://github.com/vcmi/vcmi/assets/1576820/a8782e1a-6cc5-405a-8f91-1aaa74168b24)

Reply:
Will fix it! :)

Never used software cursor

Reply:
You might want to create separate method for your dimming.
I think we fill surface with fully-transparent color for cursors, to erase any existing data.

Reply:
see  #2740